src/commands/Games/higherLower.ts

```diff
@@ -0,0 +1,267 @@
+			/** The game's reaction collector */
+			llrc: new LongLivingReactionCollector(this.client, reaction => {
+				if (game.callback === null) return;
+
+				const emoji = this.resolveCollectedEmoji(message, game, reaction);
+				if (emoji === null) return;
+
+				game.callback(emoji);
+				game.callback = null;
+			}, async () => {
+				if (game.callback !== null) {
+					game.callback(null);
+					game.callback = null;
+				}
+				try {
+					await this.end(game, message, game.turn > 1);
+				} catch (error) {
+					this.client.emit(Events.Wtf, error);
+				}
+			}),
+			response,
+			running: true,
+			turn: 1,
+			number: this.random(0),
+			wager,
+			emojis: this.kFirstReactionArray,
+			callback: null
+		};
+
+		while (game.running) {
+			await game.response.reactions.removeAll();
+			// Send the embed
+			const { TITLE, DESCRIPTION, FOOTER } = message.language.tget('COMMAND_HIGHERLOWER_EMBED');
+			await game.response.edit(null, new MessageEmbed()
+				.setColor(getColor(message))
+				.setTitle(TITLE(game.turn))
+				.setDescription(DESCRIPTION(game.number))
+				.setFooter(FOOTER));
+
+			// Add the options
+			if (game.turn > 1) game.emojis = this.kReactionArray;
+			for (const emoji of game.emojis) {
+				await game.response.react(emoji);
+			}
+
+			const emoji = await this.awaitReactionAction(game);
+			if (emoji === null) break;
+
+			// Main game logic (at last)
+			const oldNum = game.number;
+			game.number = this.random(game.number);
+
+			switch (emoji) {
+				case HigherLowerReactions.Higher: {
+					game.running = await (game.number > oldNum ? this.win(game, message) : this.loss(game, message));
+					break;
+				}
+				case HigherLowerReactions.Lower: {
+					game.running = await (game.number < oldNum ? this.win(game, message) : this.loss(game, message));
+					break;
+				}
+				case HigherLowerReactions.Cashout: {
+					await this.end(game, message, true);
+					game.running = false;
+					break;
+				}
+			}
+
+			if (game.running) game.turn++;
+		}
+	}
+
+	private async win(game: HigherLowerGameData, message: KlasaMessage) {
+		const { language } = message;
+		await game.response.reactions.removeAll();
+
+		// TODO(Quantum): Implement Winning event for InfluxDB
+		const { TITLE, DESCRIPTION, FOOTER } = message.language.tget('COMMAND_HIGHERLOWER_WIN');
+		await game.response.edit(null, new MessageEmbed()
+			.setColor(getColor(message))
+			.setTitle(TITLE)
+			.setDescription(DESCRIPTION(this.calculateWinnings(game.wager, game.turn), game.number))
+			.setFooter(FOOTER));
+
+		// Ask the user whether they want to continue or cashout
+		game.emojis = this.kWinReactionArray;
+		for (const emoji of game.emojis) {
+			await game.response.react(emoji);
+		}
+
+		const reaction = await this.awaitReactionAction(game);
+		await game.response.reactions.removeAll();
+
+		const action = reaction === null
+			? EndingAction.Timeout
+			: reaction === HigherLowerReactions.Ok
+				? EndingAction.Play
+				: EndingAction.Stop;
```

Comment:
nested ternaries are kind of ew and harder to read. If this can be refactored that would be nice.

> action is, if reaction strictly equals null, EndingAction.Timeout unless that is falsey then if reaction strictly equals HigherLowerReactions.Ok is truthy it is EndingAction.play else if false then it's EndingAction.stop

Just to put it to words to clarify why it's vague.

Reply:
> Action is, if reaction is strictly null, a timeout, otherwise if it was ok, then it'll continue the game, else it'll stop.

This works because LLRC validates all data before it's being emitted, making `reaction` one of `null | HigherLowerReactions.Ok | HigherLowerReactions.Cancel`.

Sure we can remove the ternary, I can move it to the switch, will be weird though.